cmd/hendlers/hendlers.go

```diff
@@ -0,0 +1,87 @@
+package hendlers
+
+import (
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type Subj struct {
+	//sync.Mutex
+	ID      string `json:"id"`
+	LongURL string `json:"LongURL"`
+}
+
+type JSON struct {
+	sync.Mutex
+	Obj []Subj
+}
+
+var sett JSON
+
+func BodyHandler(w http.ResponseWriter, r *http.Request) {
+	// читаем Body
+	switch r.Method {
+	case "POST":
+		b, _ := io.ReadAll(r.Body)
+
+		defer r.Body.Close()
+
+		str := string(b)
+		sitr, err := url.PathUnescape(str)
+		if err != nil {
+			fmt.Println(err)
+		}
+		id := strconv.Itoa(rand.Int())
+		strURL := "http://localhost:8080/" + id
+
+		w.WriteHeader(http.StatusCreated)
+
+		var settings JSON
+
+		newURL := Subj{
+			ID:      id,
+			LongURL: sitr,
+		}
+
+		sett.Obj = append(settings.Obj, newURL)
+
+		w.Write([]byte(strURL))
+
+		fmt.Println(sett.Obj)
+
+	default:
+		w.WriteHeader(400)
+	}
+}
+
+func GetHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
+		q := r.URL.String()
```

Comment:
```suggestion
		q := chi.URLParam(request, "urlID")
```
И не придется отрезать /